airbyte-commons-server/src/main/java/io/airbyte/commons/server/services/AirbyteRemoteOssCatalog.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.commons.server.services;
+
+import io.airbyte.config.CatalogDefinitionsConfig;
+import io.airbyte.config.init.RemoteDefinitionsProvider;
+import io.airbyte.config.specs.CombinedConnectorCatalogDownloader;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Convenience class for retrieving the remote OSS Catalog.
+ */
+@SuppressWarnings("PMD.AvoidCatchingThrowable")
+public class AirbyteRemoteOssCatalog extends RemoteDefinitionsProvider {
+
+  private static final String REMOTE_OSS_CATALOG_URL = CatalogDefinitionsConfig.getRemoteOssCatalogUrl();
+  private static final long TIMEOUT = 30000;
+  private static final Logger LOGGER = LoggerFactory.getLogger(CombinedConnectorCatalogDownloader.class);
+
+  public AirbyteRemoteOssCatalog() {
+    super(REMOTE_OSS_CATALOG_URL, TIMEOUT);
+    LOGGER.info("Initializing OSS catalog from {} with timeout {}", REMOTE_OSS_CATALOG_URL, TIMEOUT);
+  }
+
+}
```

Comment:
> I would like to have this use the RemoteDefinitionProvider singleton, however we need this to always go to OSS definitions not cloud definitions

Could you elaborate on why we need this to always go to OSS definitions? The RemoteDefinitionsProvider in use should point to the right one depending on the environment (cloud to cloud catalog, oss to oss catalog). If we ever decide to do somehting on cloud to show the latest versions, I think it makes sense for it to use the Cloud catalog as source of truth for that rather than the OSS catalog. We were hardcoding it to the YAMLs available in OSS before because these methods were only being called in OSS, but it doesn't feel like a requirement to me.

Reply:
@pedroslopez So I was going off of a comment left by Evan that I had misintterpretted.

https://github.com/airbytehq/airbyte/pull/23055#issuecomment-1433739415

I had originally understood it as we always want to point to OSS even in cloud (which is what we have been doing)

But after rereading it this morning its seems clear that what we are currently doing is not correct and we do want to reference the cloud catalog for this feature. Which means youre right we can remove this and just reference the DefinitionProvider

@evantahler Triple checking, is that correct?

Reply:
I think I'm confused.  Shouldn't the OSS platform use REMOTE_OSS_CATALOG_URL and the cloud platform use REMOTE_CLOUD_CATALOG_URL?

Reply:
Ok so this IS confusing, not just you.

For context sake, in the video below I try to explain
1. How this was working before the split
2. The assumptions/context as to why
3. How its been designed to work during the split

This has led to two questions for the product (@evantahler )
1. Should we continue to use `LocalDefinitionsProvider` in oss to support airgapped? (Like we have been)
2. `AirbyteGithubStore` always referenced the OSS version of the catalog, regardless of the context it was running in (oss/cloud), should we continue that behaviour? Or should we just use the same catalog used to seed the database?

https://www.loom.com/share/cbe547ead2b44f0086a8066c34280b30

Reply:
(Sweet loom!) My take:

1. Yes, we should continue to use LocalDefinitionsProvider in OSS. We can decide later on to bring in auto-updates and remote definitions to load connectors in OSS, but I think that's orthogonal to this work and a different conversation.
2. We should just use the same catalog used to seed the database. This was "a bug", but in practice this hasn't been an issue in the past because it's important to realize *users can't upgrade connectors on their own for cloud*. This also means we don't show the latest version on that settings page in Cloud, unlike we do in OSS.

Reply:
Another loom reply!
https://www.loom.com/share/e05991c0251d46569cd93e9020f41b05

Basically I agree 100% with Pedro but I think to change to the DI version of `DefinitionProvider` will require a larger refactor as we're currently between a migration to micronaut?

I believe this is what @colesnodgrass and I talked about.

Reply:
Thanks again for the loom!

Argh yeah, I think the in-progress micronaut migration makes this a bit harder. I'm ok with pushing this piece off until the cloud server has been migrated to micronaut, which seems like it's very close anyway. 

Regardless, I think something else you said in that loom makes me think we _shouldn't_ actually use the same global DefinitionsProvider singleton, and that's the fact that we would be using LocalDefinitionsProvider to get latest versions in OSS since we're using LocalDefinitionsProvider to load things in... _but_ we want to use a RemoteDefinitionsProvider instead to get the lateset versions. This means that we would have a LocalDefinitionsProvider (for initial loading) _and_ a RemoteDefinitionsProvider (for latest versions). We may be able to still use RemoteDefinitionsProvider directly for this, but set it up as a different "named" bean, but I don't think you'll be able to do this until the server has been moved to micronaut, anyway.

All this to say, I think we might have to keep a separate class for this for now.

Reply:
Ok! Well silver lining for me here is that I didn't misunderstand the system!

Reply:
Added a @deprectaed decorator to the class to mark this for removal on micronaut migration